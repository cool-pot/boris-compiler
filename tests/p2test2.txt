*** simple calc with variables
global a=0;
global b=1;
a=(234*b-459*a)/2;
b=2;
print a;
print b;
a=(234*b-459*a)/2;
a<->b;
print a;
print b;