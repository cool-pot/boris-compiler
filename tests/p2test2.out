> Hello from parserTester for boris compiler!
Start parse tree visualization:

[ROOT_INPUT]
    [SDD_LIST]
        [DECL_AS_SDD]
            [GLOBAL_DECL]
                [PLACEHOLDER-261]global
                [ID]a
                [PLACEHOLDER-293]=
                [SINGLE_INT_AS_EXPR]
                    [INT]0
                [PLACEHOLDER-290];
        [SDD_LIST]
            [DECL_AS_SDD]
                [GLOBAL_DECL]
                    [PLACEHOLDER-261]global
                    [ID]b
                    [PLACEHOLDER-293]=
                    [SINGLE_INT_AS_EXPR]
                        [INT]1
                    [PLACEHOLDER-290];
            [SDD_LIST]
                [STATEMENT_AS_SDD]
                    [LHS_ASSIGN_EXPR_AS_STATEMENT]
                        [LHS]
                            [SINGLE_ID_AS_LHSITEM]
                                [ID]a
                            [EMPTY_NODE]
                        [PLACEHOLDER-293]=
                        [EXPR_DIV_EXPR]
                            [LPAR_EXPR_LPAR]
                                [PLACEHOLDER-291](
                                [EXPR_MINUS_EXPR]
                                    [EXPR_MULT_EXPR]
                                        [SINGLE_INT_AS_EXPR]
                                            [INT]234
                                        [PLACEHOLDER-286]*
                                        [SINGLE_ID_AS_EXPR]
                                            [ID]b
                                    [PLACEHOLDER-285]-
                                    [EXPR_MULT_EXPR]
                                        [SINGLE_INT_AS_EXPR]
                                            [INT]459
                                        [PLACEHOLDER-286]*
                                        [SINGLE_ID_AS_EXPR]
                                            [ID]a
                                [PLACEHOLDER-292])
                            [PLACEHOLDER-287]/
                            [SINGLE_INT_AS_EXPR]
                                [INT]2
                        [PLACEHOLDER-290];
                [SDD_LIST]
                    [STATEMENT_AS_SDD]
                        [LHS_ASSIGN_EXPR_AS_STATEMENT]
                            [LHS]
                                [SINGLE_ID_AS_LHSITEM]
                                    [ID]b
                                [EMPTY_NODE]
                            [PLACEHOLDER-293]=
                            [SINGLE_INT_AS_EXPR]
                                [INT]2
                            [PLACEHOLDER-290];
                    [SDD_LIST]
                        [STATEMENT_AS_SDD]
                            [PRINT_STATEMENT]
                                [PLACEHOLDER-274]print
                                [SINGLE_ID_AS_EXPR]
                                    [ID]a
                                [PLACEHOLDER-290];
                        [SDD_LIST]
                            [STATEMENT_AS_SDD]
                                [PRINT_STATEMENT]
                                    [PLACEHOLDER-274]print
                                    [SINGLE_ID_AS_EXPR]
                                        [ID]b
                                    [PLACEHOLDER-290];
                            [SDD_LIST]
                                [STATEMENT_AS_SDD]
                                    [LHS_ASSIGN_EXPR_AS_STATEMENT]
                                        [LHS]
                                            [SINGLE_ID_AS_LHSITEM]
                                                [ID]a
                                            [EMPTY_NODE]
                                        [PLACEHOLDER-293]=
                                        [EXPR_DIV_EXPR]
                                            [LPAR_EXPR_LPAR]
                                                [PLACEHOLDER-291](
                                                [EXPR_MINUS_EXPR]
                                                    [EXPR_MULT_EXPR]
                                                        [SINGLE_INT_AS_EXPR]
                                                            [INT]234
                                                        [PLACEHOLDER-286]*
                                                        [SINGLE_ID_AS_EXPR]
                                                            [ID]b
                                                    [PLACEHOLDER-285]-
                                                    [EXPR_MULT_EXPR]
                                                        [SINGLE_INT_AS_EXPR]
                                                            [INT]459
                                                        [PLACEHOLDER-286]*
                                                        [SINGLE_ID_AS_EXPR]
                                                            [ID]a
                                                [PLACEHOLDER-292])
                                            [PLACEHOLDER-287]/
                                            [SINGLE_INT_AS_EXPR]
                                                [INT]2
                                        [PLACEHOLDER-290];
                                [SDD_LIST]
                                    [STATEMENT_AS_SDD]
                                        [LHS_EXCHANGE_LHS_AS_STATEMENT]
                                            [LHS]
                                                [SINGLE_ID_AS_LHSITEM]
                                                    [ID]a
                                                [EMPTY_NODE]
                                            [PLACEHOLDER-294]<->
                                            [LHS]
                                                [SINGLE_ID_AS_LHSITEM]
                                                    [ID]b
                                                [EMPTY_NODE]
                                            [PLACEHOLDER-290];
                                    [SDD_LIST]
                                        [STATEMENT_AS_SDD]
                                            [PRINT_STATEMENT]
                                                [PLACEHOLDER-274]print
                                                [SINGLE_ID_AS_EXPR]
                                                    [ID]a
                                                [PLACEHOLDER-290];
                                        [SDD_LIST]
                                            [STATEMENT_AS_SDD]
                                                [PRINT_STATEMENT]
                                                    [PLACEHOLDER-274]print
                                                    [SINGLE_ID_AS_EXPR]
                                                        [ID]b
                                                    [PLACEHOLDER-290];
                                            [EMPTY_NODE]
